### PR TITLE
Update CI integration, switch Cygwin tests to GH Actions

### PR DESCRIPTION
**Pull request by fingolfin**
Please DO NOT MERGE until I've verified it all works as expected

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -15,41 +16,78 @@ concurrency:
...
+          ABI: ${{ matrix.ABI }}
       - uses: gap-actions/build-pkg@v1
       - uses: gap-actions/run-pkg-tests@v2
       - uses: gap-actions/process-coverage@v2
-      - uses: codecov/codecov-action@v2
+      - uses: codecov/codecov-action@v3
+
+  test-cygwin:
+    name: "master cygwin64"
+    if: ${{ !(github.event_name == 'pull_request' && github.event.pull_request.head.repo.full_name == github.repository) }}
```
Comment (fingolfin) [inline .github/workflows/CI.yml:62]:
```suggestion
    # if: ${{ !(github.event_name == 'pull_request' && github.event.pull_request.head.repo.full_name == github.repository) }}
```

**Comment by lhsoicher** [comment]
@fingolfin Please let me know if/when I can merge this PR. Thanks.

**Comment by fingolfin** [comment]
Strange, it dies with error
```
+ ./configure /home/runneradmin/gap
./configure: line 4: /home/runneradmin/gap/sysinfo.gap
```
inside of `gap-actions/build-pkg@cygwin-v1`; yet in the preceding `gap-actions/setup-gap@cygwin-v2` that file existed and seemingly worked fine:
```
  ==== Checking profiling
  Running './configure' '/home/runneradmin/gap' 
  Using settings from /home/runneradmin/gap/sysinfo.gap
  Running 'gmake' 'clean' 
  rm -rf bin/x86_64-pc-cygwin-default64-kv8 gen
  Running './configure' '/home/runneradmin/gap' 
  Using settings from /home/runneradmin/gap/sysinfo.gap
  Running 'gmake' 
     GAC     src/profiling.cc => gen/src/profiling.o
     GAC     src/md5.cc => gen/src/md5.o
     GAC     gen/src/profiling.o => bin/x86_64-pc-cygwin-default64-kv8/profiling.so

```

@ChrisJefferson @wilfwilson any idea what might be causing this?

**Comment by wilfwilson** [comment]
I'm really not sure @fingolfin, sorry.

**Comment by fingolfin** [comment]
This is really strange, considering that I copied the CI setup basically verbatim from `datastructures` and `io`, where it works smoothly.

Perhaps this requires using tmate to log into the GH action CI job and investigate more closely. I'll try and see if I can do that (if nobody beats me to it) later today, but now I need to prepare for my lecture

**Comment by ChrisJefferson** [comment]
I will have a quick look now

**Comment by ChrisJefferson** [comment]
One problem is the `./configure` script uses `source`, but hard-wires `/bin/sh`, so systems that don't use `bash` as `sh` won't work (I'm surprised this hasn't come up before!)

**Comment by ChrisJefferson** [comment]
One issue is you've changed the `configure` script to use `source` instead of `.`, but the script is hard-wired to `/bin/sh`, which doesn't (have to) support `source` if it isn't actually bash.

**Comment by fingolfin** [comment]
Problem was caused by linenedings. Chris fixed it by adding a `.gitatributes` file.

**Pull request closed without merging**